erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -45,6 +47,25 @@ def validate(self):
 			self.validate_supplier_invoice()
 			self.validate_advance_jv("Purchase Order")
 
+		# validate cash purchase
+		if (self.is_paid == 1):
+			self.validate_cash()
+
+		# validate stock items
+		if (self.update_stock == 1):
+			pc_obj = frappe.get_doc('Purchase Common')
+			pc_obj.validate_for_items(self)
+			
+			self.validate_purchase_return()
+			self.validate_rejected_warehouse()
+			self.validate_accepted_rejected_qty()
+
+		# sub-contracting
+		# self.validate_for_subcontracting()
+		# self.create_raw_materials_supplied("supplied_items")
+		# self.set_landed_cost_voucher_amount()
+		# self.update_valuation_rate("items")
```

Comment:
Why all these functions are commented? All the functionalities should be covered.